torchrl/envs/custom/chess.py

```diff
@@ -132,41 +205,126 @@ def _is_done(self, board):
 
     def _reset(self, tensordict=None):
         fen = None
+        pgn = None
         if tensordict is not None:
-            fen = self._get_fen(tensordict).data
-            dest = tensordict.empty()
+            if self.include_fen:
+                fen = self._get_fen(tensordict).data
+                dest = tensordict.empty()
+            if self.include_pgn:
+                fen = self._get_pgn(tensordict).data
+                dest = tensordict.empty()
         else:
             dest = TensorDict()
 
-        if fen is None:
+        if fen is None and pgn is None:
             self.board.reset()
-            fen = self.board.fen()
+            if self.include_fen and fen is None:
+                fen = self.board.fen()
+            if self.include_pgn and pgn is None:
+                pgn = self._PNG_RESTART
         else:
-            self.board.set_fen(fen)
-            if self._is_done(self.board):
-                raise ValueError(
-                    "Cannot reset to a fen that is a gameover state." f" fen: {fen}"
-                )
-
-        hashing = hash(fen)
+            if fen is not None:
+                self.board.set_fen(fen)
+                if self._is_done(self.board):
+                    raise ValueError(
+                        "Cannot reset to a fen that is a gameover state." f" fen: {fen}"
+                    )
+            elif pgn is not None:
+                self.board = self._pgn_to_board(pgn)
 
         self._set_action_space()
         turn = self.board.turn
-        return dest.set("fen", fen).set("hashing", hashing).set("turn", turn)
+        if self.include_san:
+            dest.set("san", "[SAN][START]")
+        if self.include_fen:
+            if fen is None:
+                fen = self.board.fen()
+            dest.set("fen", fen)
+        if self.include_pgn:
+            if pgn is None:
+                pgn = self._board_to_pgn(self.board)
+            dest.set("pgn", pgn)
+        dest.set("turn", turn)
+        if self.pixels:
+            dest.set("pixels", self._get_tensor_image(board=self.board))
+        return dest
+
+    _cairosvg_lib = None
+
+    @_classproperty
+    def _cairosvg(cls):
+        csvg = cls._cairosvg_lib
+        if csvg is None:
+            import cairosvg
+
+            csvg = cls._cairosvg_lib = cairosvg
+        return csvg
+
+    _torchvision_lib = None
+
+    @_classproperty
+    def _torchvision(cls):
+        tv = cls._torchvision_lib
+        if tv is None:
+            import torchvision
+
+            tv = cls._torchvision_lib = torchvision
+        return tv
+
+    @classmethod
+    def _get_tensor_image(cls, board):
+        try:
+            svg = board._repr_svg_()
```

Comment:
Maybe we could have another option that has a more minimal representation as well, if we want to train without vision. For instance, OpenSpiel's chess env has a representation where each square of the board is given a size 20 array of ones and zeros, which is somehow used to represent which piece is in that square and which color it is

Reply:
Sure, I was using this because it's builtin - but a more minimal representation could also be cool!